### PR TITLE
pdsh: Fix building on aarch64-darwin

### PR DESCRIPTION
**Pull request by cransom**
The build is currently broken when compiling on ARM Darwin devices due to an `ld: symbol(s) not found for architecture arm64` error. Running autoreconfHook resolves this.

###### Description of changes
Runs autoreconfHook and patches `configure.ac` to remove a git command that derives the version.

###### Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [x] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [22.11 Release Notes (or backporting 22.05 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2211-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
  - [ ] (Release notes changes) Ran `nixos/doc/manual/md-to-db.sh` to update generated release notes
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

File: pkgs/tools/networking/pdsh/default.nix [hunk 1]
```diff
@@ -9,9 +9,14 @@ stdenv.mkDerivation rec {
     sha256 = "1s91hmhrz7rfb6h3l5k97s393rcm1ww3svp8dx5z8vkkc933wyxl";
   };
 
-  buildInputs = [ perl readline ssh ]
+  buildInputs = [ perl readline ssh autoreconfHook ]
```
Comment (bjornfor) [inline pkgs/tools/networking/pdsh/default.nix:12]: I think autoreconfHook goes in nativeBuildInputs.

File: pkgs/tools/networking/pdsh/default.nix [hunk 2]
```diff
@@ -9,9 +9,14 @@ stdenv.mkDerivation rec {
...
     ++ (lib.optional slurmSupport slurm);
 
+  # Do not use git to derive a version.
+  prePatch = ''
```
Comment (SuperSandro2000) [inline pkgs/tools/networking/pdsh/default.nix:16]:
```suggestion
  postPatch = ''
```

**Comment by cransom** [comment]
thanks for the suggestions. Updated.

File: pkgs/tools/networking/pdsh/default.nix [hunk 3]
```diff
@@ -12,6 +12,13 @@ stdenv.mkDerivation rec {
   buildInputs = [ perl readline ssh ]
     ++ (lib.optional slurmSupport slurm);
 
+  nativeBuildInputs = [ autoreconfHook ];
+
+  # Do not use git to derive a version.
+  postPatch = ''
+    sed -i 's/m4_esyscmd(\[git describe.*/[${version}])/' configure.ac
```
Comment (bjornfor) [inline pkgs/tools/networking/pdsh/default.nix:19]: This postPatch hunk seems unrelated to the commit topic. Split into a new commit?
